codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMapAggregator.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.CodecException;
+import io.netty.handler.codec.MessageToMessageDecoder;
+import io.netty.util.ReferenceCountUtil;
+import io.netty.util.internal.UnstableApi;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Aggregates {@link RedisMessage} parts into {@link MapRedisMessage}.
+ * This decoder should be used together with {@link RedisDecoder}.
+ */
+@UnstableApi
+public final class RedisMapAggregator extends MessageToMessageDecoder<RedisMessage> {
+
+    private final Deque<AggregateState> depths = new ArrayDeque<AggregateState>(4);
```

Comment:
I feel like you need to implemented `handlerRemoved(...)` and ensure you call `release()` on each message that was added to the state.

Reply:
Does `RedisArrayAggregator` need to make the same modification？

Reply:
Only `AbstractCollectionRedisMessage` and `AbstractMapRedisMessage` extends `AbstractReferenceCounted`. 
So, other common redis message such as `bulkString` don't need call `release()`?

And, The release refCnt of `CollectionRedisMessage` and `MapRedisMessage` in override `deallocate()` of `AbstractCollectionRedisMessage` and `AbstractMapRedisMessage`